core/jobs/batch_jobs/test_gcs_io_job.py

```diff
@@ -0,0 +1,200 @@
+from core.platform import models
+
+import apache_beam as beam
+from apache_beam.io.gcp import gcsio_test
+from typing import Optional
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class TestGCSIoWriteJob(base_jobs.JobBase):
+    """Write dummy files to GCS."""
+
+    def __init__(
+        self,
+        pipeline: beam.Pipeline,
+        client: Optional[gcsio_test.FakeGcsClient] = None
+    ) -> None:
+        super().__init__(pipeline=pipeline)
+        self.client = client
+        self.pipeline = pipeline
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        write_files_to_gcs = (
+            self.pipeline
+            | 'Create PCollection having test bucket and test files' >> (
+                beam.Create(
+                    [
+                        {
+                            'filepath': (
+                                'dummy_folder/dummy_subfolder/dummy_file_1'),
+                            'data': b'testing_1'
+                        },
+                        {
+                            'filepath': (
+                                'dummy_folder/dummy_subfolder/dummy_file_2'),
+                            'data': b'testing_2'
+                        }
+                    ]
+                ))
+            | 'Write files to GCS' >> gcs_io.WriteFile(self.client)
+        )
+
+        total_files_write = (
+            write_files_to_gcs
+            | 'Total number of files write to GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL FILES WRITTEN'))
+        )
+
+        return total_files_write
+
+
+class TestGCSIoReadJob(base_jobs.JobBase):
+    """Read dummy files from GCS."""
+
+    def __init__(
+        self,
+        pipeline: beam.Pipeline,
+        client: Optional[gcsio_test.FakeGcsClient] = None
+    ) -> None:
+        super().__init__(pipeline=pipeline)
+        self.client = client
+        self.pipeline = pipeline
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        read_files_from_gcs = (
+            self.pipeline
+            | 'Create PCollection of files that needs to be fetched' >> (
+                beam.Create(
+                    [
+                        'dummy_folder/dummy_subfolder/dummy_file_1',
+                        'dummy_folder/dummy_subfolder/dummy_file_2'
+                    ]
+                ))
+            | 'Read files from the GCS' >> gcs_io.ReadFile(self.client, mode='r')
+        )
+
+        total_files_read = (
+            read_files_from_gcs
+            | 'Total number of files read from GCS' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'TOTAL FILES FETCHED'))
+        )
+
+        output_files = (
+            read_files_from_gcs
+            | 'Output the data' >> beam.Map(lambda data: (
+                job_run_result.JobRunResult.as_stdout(f'The data is {data}')
+            ))
```

Comment:
The data for file ### is ###

Reply:
I'm a bit confused on how to map this part with files, I will update this part after discussing with you.
Thanks

Reply:
Hmm, maybe the `ReadFile` should return tuple of name and data?